src/kernels/webgl/conv_gpu_depthwise_cs.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {Conv2DInfo} from '../../ops/conv_util';
+import {GPGPUProgram} from './gpgpu_math';
+
+export class DepthwiseConv2DProgramCS implements GPGPUProgram {
+  variableNames = ['x', 'W'];
+  outputShape: number[];
+  userCode: string;
+  localGroupSize: [number, number];
+
+  constructor(convInfo: Conv2DInfo) {
+    this.outputShape = convInfo.outShape;
+    const xNumRows = convInfo.inHeight;
+    const xNumCols = convInfo.inWidth;
+    const padTop = convInfo.padInfo.top;
+    const padLeft = convInfo.padInfo.left;
+    const strideHeight = convInfo.strideHeight;
+    const strideWidth = convInfo.strideWidth;
+    const dilationHeight = convInfo.dilationHeight;
+    const dilationWidth = convInfo.dilationWidth;
+    const filterHeight = convInfo.filterHeight;
+    const filterWidth = convInfo.filterWidth;
+    const channelMul = convInfo.outChannels / convInfo.inChannels;
+
+    this.localGroupSize = [8 * channelMul, 7];
+    // outWidth should be divisible by localGroupSize[1]
+
+    this.userCode = `
+      const ivec2 strides = ivec2(${strideHeight}, ${strideWidth});
+      const ivec2 pads = ivec2(${padTop}, ${padLeft});
+
+      const int CACHE_H = ${filterHeight};
+      const int CACHE_W = ${
+        (this.localGroupSize[1] - 1) * strideWidth + filterWidth +
+        (filterWidth - 1) * (dilationWidth - 1)};
+      const int CACHE_C = ${this.localGroupSize[0] / channelMul};
```

Comment:
Why use `CACHE_C = ${this.localGroupSize[0] / channelMul}` not `CACHE_C = ${convInfo.outChannels / channelMul}` ?

Reply:
![depthiseConv](https://user-images.githubusercontent.com/35633199/57188127-07b41f80-6f2c-11e9-9fa9-8a35df76a365.png)
I drew a simple picture which may help to understand.

If `localGroupSize[0] < outChannels`, we don't need to load data from all channels.
But if `localGroupSize[0] >= outChannels`, it's a good advice to set `CACHE_C = ${convInfo.outChannels / channelMul}`.